### PR TITLE
add new `viz` submodule

### PR DESCRIPTION
**Pull request by ryanSoley**
this branch will serve as the staging ground for the new visualization work

## What
  * updates `dash` to 2.0.0 and `dash-bootstrap-components` to 1.0.0
    * issues between updated dash libraries seems to be stable now
    * the new major releases are very much not backwards compatible - might as well upgrade sooner than later
  * adds new widgets to the `viz` submodule
    * dataframe plotting
    * experiments table
    * list metric comparison
    * metric correlation plotting
  * adds dashboarding capabilities
  * adds example notebooks for new widgets
    * https://github.com/capitalone/rubicon-ml/tree/viz-prototypes/notebooks/viz

## To Do
- [x] #146
- [x] #147
- [x] harden dashboard capabilities
- [x] test new code and reenable the tests that were disabled in this PR

to do at a later date after this PR is merged:
* #148
* #161 
* #165 

## How to Test
  * create a new dev environment
    * `mamba env create -f environment.yml`
  * run through the new notebook examples
    * https://github.com/capitalone/rubicon-ml/tree/viz-prototypes/notebooks/viz

### CODE REVIEW

File: notebooks/viz/metric-comparisons.ipynb [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{
```
Comment (andreafehrman) [inline notebooks/viz/metric-comparisons.ipynb:1]: When bringing up the visualization, it looks like there are only variable names with values right under them. The rest is just empty space in the table. What is the intended visualization for this notebook?

File: rubicon_ml/viz/metric_lists_comparison.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            x=column_names if len(column_names) == len(heatmap_data[0]) else None,
+            y=experiment_ids,
+        )
+        heatmap.update_layout(
+            margin_b=30, margin_t=30, modebar_orientation="v", plot_bgcolor=plot_background_blue
+        )
+        heatmap.update_xaxes(gridcolor="white")
+        heatmap.update_yaxes(gridcolor="white")
+
+        heatmap_cell_rem = 6
```
Comment (shania-m) [inline rubicon_ml/viz/metric_lists_comparison.py:129]: these numbers are hardcoded, could we have them stored as variables

File: rubicon_ml/viz/metric_lists_comparison.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        heatmap_height = 12 + (len(heatmap_data) * (heatmap_cell_rem / 2))
```
Comment (shania-m) [inline rubicon_ml/viz/metric_lists_comparison.py:130]: same as above

File: rubicon_ml/viz/metric_lists_comparison.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        heatmap_width = (
+            12 + (len(heatmap_data[0]) * heatmap_cell_rem) if len(heatmap_data[0]) > 8 else 72
```
Comment (shania-m) [inline rubicon_ml/viz/metric_lists_comparison.py:132]: same as above

File: notebooks/viz/dashboard-compilation.ipynb [hunk 5]
```diff
@@ -0,0 +1,116 @@
+{
```
[the remark this answers is not in the document]
Comment (ryanSoley) [reply notebooks/viz/dashboard-compilation.ipynb:1]: yeah this example will be reworked a lot before we merge anything in

File: notebooks/viz/plot-artifacts.ipynb [hunk 6]
```diff
@@ -0,0 +1,94 @@
+{
```
[the remark this answers is not in the document]
Comment (ryanSoley) [reply notebooks/viz/plot-artifacts.ipynb:1]: yeah we should probably add that option. the dataframe plotting widget allows for stacking or separate plots, so we should be consistent. we'll see if that enhancement will fall under this initial scope of work or future stuff - depends if its as trivial as simply lifting it from the dataframe plotting widget or not

File: notebooks/viz/dashboard-compilation.ipynb (hunk 5 above)
[the remark this answers is not in the document]
Comment (ryanSoley) [reply notebooks/viz/dashboard-compilation.ipynb:1]: I'll take care of this in #161

File: notebooks/viz/dashboard-compilation.ipynb [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+ "cells": [
```
[the remark this answers is not in the document]
Comment (ryanSoley) [reply notebooks/viz/dashboard-compilation.ipynb:2]: I think that'd be a good addition for a followup once this is in. I just wanted to focus on the interaction between the selected rows and the other widgets for this first pass at things

File: notebooks/viz/experiments-table.ipynb [hunk 8]
```diff
@@ -0,0 +1,79 @@
+{
+ "cells": [
```
[the remark this answers is not in the document]
Comment (ryanSoley) [reply notebooks/viz/experiments-table.ipynb:2]: yeah thats just a weird thing with the `dash_table`. I think Joe messed with it a while ago and couldn't really do anything about it so we've just left it

File: rubicon_ml/viz/experiments_table.py [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+
+        return html.Div(
+            [
+                *header_row,
+                dcc.Loading(experiment_table, color=light_blue),
+            ],
+        )
+
+    def load_experiment_data(self):
+        self.experiment_records = []
```
Comment (shania-m) [inline rubicon_ml/viz/experiments_table.py:176]: doc strings

File: rubicon_ml/viz/experiments_table.py [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+            ]
+
+        return html.Div(
+            [
+                *header_row,
+                dcc.Loading(experiment_table, color=light_blue),
+            ],
+        )
+
+    def load_experiment_data(self):
```
Comment (shania-m) [inline rubicon_ml/viz/experiments_table.py:175]: seems like these functions are doing a lot, can we get doc strings?

**Review by shania-m — APPROVED** [review]
LGTM

**Pull request merged**